lib/ansible/executor/task_executor.py

```diff
@@ -910,9 +910,9 @@ def _poll_async_result(self, result, templar, task_vars=None):
                 # have issues which result in a half-written/unparseable result
                 # file on disk, which manifests to the user as a timeout happening
                 # before it's time to timeout.
-                if (int(async_result.get('finished', 0)) == 1 or
-                        ('failed' in async_result and async_result.get('_ansible_parsed', False)) or
-                        'skipped' in async_result):
+                if (async_result.get('finished', False) or
```

Comment:
This is definitely a better version of the conditional, but the explicit `False` defaults in `dict.get()` here make it much less readable IMO- since none of them are being captured, the implicit `None` defaults (falsey) are fine.